optuna/integration/pytorch_distributed.py

```diff
@@ -24,6 +25,45 @@
 )
 
 
+def updates_properties(f: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
@himkt mentioned IDEs cannot infer argument types of the annotated methods. Could you use `ParamSpec` as can be seen in https://github.com/optuna/optuna/pull/3270?